internal/k8s/validator.go

```diff
@@ -371,6 +391,113 @@ func (l *leader) ValidateServiceImportObj(key string, serviceImport *akov1alpha1
 	return nil
 }
 
+// ValidateOAuthSamlConfigObj would do validation checks
+// update internal CRD caches, and push relevant ingresses to ingestion
+func (l *leader) ValidateOAuthSamlConfigObj(key string, oauthSamlConfig *akov1alpha2.OAuthSamlConfig) error {
+	var err error
+	fqdn := *oauthSamlConfig.Spec.Fqdn
+	foundHost, foundOSC := objects.SharedCRDLister().GetFQDNToOAuthSamlConfigMapping(fqdn)
+	if foundHost && foundOSC != oauthSamlConfig.Namespace+"/"+oauthSamlConfig.Name {
+		err = fmt.Errorf("duplicate fqdn %s found in %s", fqdn, foundOSC)
+		status.UpdateOAuthSamlConfigStatus(key, oauthSamlConfig, status.UpdateCRDStatusOptions{Status: lib.StatusRejected, Error: err.Error()})
+		return err
+	}
+
+	refData := make(map[string]string)
+
+	if oauthSamlConfig.Spec.SsoPolicyRef == nil {
+		err = fmt.Errorf("SsoPolicyRef is not specified")
+		status.UpdateOAuthSamlConfigStatus(key, oauthSamlConfig, status.UpdateCRDStatusOptions{Status: lib.StatusRejected, Error: err.Error()})
+		return err
+	}
+	refData[*oauthSamlConfig.Spec.SsoPolicyRef] = "SSOPolicy"
+
+	if oauthSamlConfig.Spec.OauthVsConfig != nil {
+		oauthConfigObj := oauthSamlConfig.Spec.OauthVsConfig
+
+		if len(oauthConfigObj.OauthSettings) != 0 {
+			for _, profile := range oauthConfigObj.OauthSettings {
+				refData[*profile.AuthProfileRef] = "AuthProfile"
+
+				if profile.AppSettings != nil {
+					clientSecret := *profile.AppSettings.ClientSecret
+					clientSecretObj, err := validateSecretReferenceInOAuthSamlConfig(oauthSamlConfig.Namespace, clientSecret)
+					if err != nil {
+						err = fmt.Errorf("Got error while fetching %s secret : %s", clientSecret, err.Error())
+						status.UpdateOAuthSamlConfigStatus(key, oauthSamlConfig, status.UpdateCRDStatusOptions{Status: lib.StatusRejected, Error: err.Error()})
+						return err
+					}
+					if clientSecretObj == nil {
+						err = fmt.Errorf("specified client secret is empty : %s", clientSecret)
+						status.UpdateOAuthSamlConfigStatus(key, oauthSamlConfig, status.UpdateCRDStatusOptions{Status: lib.StatusRejected, Error: err.Error()})
+						return err
+					}
+					clientSecretString := string(clientSecretObj.Data["clientSecret"])
+					if clientSecretString == "" {
+						err = fmt.Errorf("clientSecret field not found in %s secret", clientSecret)
+						status.UpdateOAuthSamlConfigStatus(key, oauthSamlConfig, status.UpdateCRDStatusOptions{Status: lib.StatusRejected, Error: err.Error()})
+						return err
+					}
+				}
+
+				if profile.ResourceServer != nil {
+					if *profile.ResourceServer.AccessType == "ACCESS_TOKEN_TYPE_JWT" && profile.ResourceServer.JwtParams == nil {
```

Comment:
a nil check seems to be missing for the AccessType or is it having a default value?

Reply:
AccessType is a required field in the CRD, so nil check is not required.